libs/ledger/include/ledger/protocols/main_chain_rpc_service.hpp

```diff
@@ -127,7 +132,10 @@ class MainChainRpcService : public muddle::rpc::Server,
   /// @{
   static constexpr char const *ToString(State state) noexcept;
   Address                      GetRandomTrustedPeer() const;
-  void                         HandleChainResponse(Address const &address, BlockList block_list);
+
+  void HandleChainResponse(Address const &address, BlockList blocks);
```

Comment:
It seems like the OnWaitForHeaviestChain code is still there, which is why you are templating on HandleChainResponse - can we not delete the COMMON_SUB_CHAIN call in OnSyncronising? It might be possible that this would hide any errors in the forward sync by effectively doing the correct thing when miners are all close to HEAD